test/e2e/ratelimit/ratelimit_universal.go

```diff
@@ -0,0 +1,136 @@
+package ratelimit
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/config/core"
+	. "github.com/kumahq/kuma/test/framework"
+)
+
+func RateLimitOnUniversal() {
+	var cluster Cluster
+	var deployOptsFuncs []DeployOptionsFunc
+	rateLimitPolicy := `
+type: RateLimit
+mesh: default
+name: rate-limit-all-sources
+sources:
+- match:
+   kuma.io/service: "*"
+destinations:
+- match:
+   kuma.io/service: echo-server_kuma-test_svc_8080
+conf:
+ http:
+   connections: 5
+   interval: 10s
+   onError:
+     status: 423
+     headers:
+       - key: "x-kuma-rate-limited"
+         value: "true"
+`
+
+	BeforeEach(func() {
```

Comment:
Please write this test with BeforeSuite / AfterSuite. We already learned that writing it this way is a huge win for test performance and writing new test cases.